Events.md

```diff
@@ -0,0 +1,225 @@
+# Introduction
+
+Events in solidity is a hard thing to do in a optimal way. If you look at it as a purely gas efficient issue, you want to emit as few events as possible and with as few fields as possible. But as events also has to be usable for an off chain user, then doing this is not always the best solution.
+
+We tried to list the main points that we would like to do with events.
+
+1. Use as little gas as possible
+2. An indexer should be able to keep track of the state of Mangrove.
+3. Doing RPC calls directly, should be able to find offers and other information based on offerlist, maker, taker, offer id, etc.
+
+These 3 points all have there own direction and it is therefore not possible to find a solution that is optimal for all 3 points.
+We have therefore tried to find a solution that is a good balance between the all 3 points.
+
+## Events
```

Comment:
IMHO all the event-specific documentation should be inlined where the events are defined. @adhusson wdyt?

Reply:
All the documentation has been inlined at their relevant event.